integration-cli/docker_cli_service_create_test.go

```diff
@@ -90,35 +91,128 @@ func (s *DockerSwarmSuite) TestServiceCreateWithSecretSimple(c *check.C) {
 	c.Assert(refs[0].File.Name, checker.Equals, testName)
 	c.Assert(refs[0].File.UID, checker.Equals, "0")
 	c.Assert(refs[0].File.GID, checker.Equals, "0")
+
+	out, err = d.Cmd("service", "rm", serviceName)
+	c.Assert(err, checker.IsNil, check.Commentf(out))
+	d.DeleteSecret(c, testName)
 }
 
-func (s *DockerSwarmSuite) TestServiceCreateWithSecretSourceTarget(c *check.C) {
+func (s *DockerSwarmSuite) TestServiceCreateWithSecretSourceTargetPaths(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	testPaths := map[string]string{
+		"app":             "/etc/secret",
+		"test_secret":     "test_secret",
+		"relative_secret": "relative/secret",
+		"invalid_secret":  "../secret",
```

Comment:
Would checking `path == filepath.Clean(path)` work across platform, and just clean any references to `..`?  It might be more explicit to fail this way.

If I understand correctly, this particular case is invalid because on the host it would take you out of the flat mount directory, but something like `ignoreddir/../test_secret` would be fine because that would end up being just `secret`.  That might be confusing to users, since they may just want it in `/var/run/test_secret` as opposed to `/var/run/secrets/test_secret`, and may not understand that the `../` is disallowed because it's taking them out of the host path specifically.